packages/svelte2tsx/src/svelte2tsx/index.ts

```diff
@@ -281,24 +281,25 @@ function addComponentExport({
     fileName,
     componentDocumentation
 }: AddComponentExportPara) {
-    const eventsDef = strictEvents ? 'render' : '__sveltets_with_any_event(render)';
+    // FIXME: Does this make sense? Are we supposed to be able to listen to events our component does not emit?
+    const eventsDef = "ReturnType<typeof render>['events']" + (strictEvents ? '' : ' & Record<string, CustomEvent<any>>');
```

Comment:
Yes this is on purpose. Image you have some dispatcher mixin imported into the component which emits events as well. There is no way to find out from just looking at the Svelte file.
Also this code contains an error, the generic is not closed (`>`) if `strictEvents` is `true`.

Reply:
> Also this code contains an error, the generic is not closed
I am counting an equal number of opening and closing tags.   `<>` + (strict events ? `` : `<<>>`)

Reply:
Yeah sorry, my bad 😄